c_code.cpp

```
@@ -0,0 +1,761 @@
+{
+    if (s_cellularif_conf.enable_bind_sim && s_cellularif_conf.bind_iccid == NULL &&
+        new_status->iccid != NULL) {
+        log_debug("networkd", "", "[info] SIM lazy binding, bind with: %s\n", new_status->iccid);
+        s_cellularif_conf.bind_iccid = strdup(new_status->iccid);
+        write_iccid(s_cellularif_conf.bind_iccid);
+    }
+
+    if (s_cellularif_conf.enable_bind_sim && s_cellularif_conf.bind_iccid != NULL &&
+        new_status->iccid != NULL) {
+        if (strncmp(s_cellularif_conf.bind_iccid, new_status->iccid,
+                    strlen(s_cellularif_conf.bind_iccid) != 0)) {
```

cppcheck (2.10-2)

[error] Invalid strncmp() argument nr 3. A non-boolean value is required.

(invalidFunctionArgBool)